File: Baekjoon/step by step/08_math_1/11005.py
# 11005: 진법 변환 2

digit_dict = { i+10: chr(i+65) for i in range(26) }
base_B = ""

# 1. 입력 받기
N, B = list(map(int, input().split()))

# 2. base_N 진법으로 변환
while (N > 0):
    base_B = (digit_dict[N%B] if N%B >= 10 else str(N%B)) + base_B
    N //= B

print(base_B)

File: Baekjoon/step by step/08_math_1/11005.cpp
// 11005: 진법 변환 2

#include <iostream>
#include <string>
#include <unordered_map>

int main() {
  std::unordered_map<int, std::string> digit_dict = {};
  for (int i=0; i<26; i++) {
    digit_dict[i+10] = char(i+65);
  }
  
  std::string base_B = "";
  
  int N, B;
  std::cin >> N >> B;

  while (N > 0) {
    base_B = (N%B >= 10 ? digit_dict[N%B] : std::to_string(N%B)) + base_B;
    N /= B;
  }

  std::cout << base_B;

  return 0;
}